dropwizard-db/src/main/java/io/dropwizard/db/AbstractDataSourceFactory.java

```diff
@@ -0,0 +1,66 @@
+package io.dropwizard.db;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.base.Optional;
+import io.dropwizard.util.Duration;
+
+import java.util.Map;
+
+/**
+ * Interface of an abstract JDBC connection pool
+ */
+public interface AbstractDataSourceFactory {
+
+    /**
+     * Whether ORM tools allowed to add comments to SQL queries
+     *
+     * @return {@code true}, if allowed
+     */
+    boolean isAutoCommentsEnabled();
+
+    /**
+     * Gets the configuration properties for ORM tools.
+     *
+     * @return configuration properties as a map
+     */
+    Map<String, String> getProperties();
+
+    /**
+     * Gets the timeout for awaiting a response from the database.
+     * during health checks.
+     *
+     * @return the timeout as {@code Duration}
+     */
+    Optional<Duration> getHealthCheckValidationTimeout();
+
+    /**
+     * Gets the SQL query, that used for checking the database availability.
```

Comment:
Should probably read "Returns the SQL query which is being used for the database connection health check.".